### PR TITLE
feat: add secp256k1 diddoc and verification method

### PR DESCRIPTION
**Pull request by sairanjit**
### Summary

- Added `keyDidSecp256k1`
- Added Verification method Type `EcdsaSecp256k1VerificationKey2019`
- Added a `test-vector` from `https://github.com/w3c-ccg/did-method-key/blob/main/test-vectors/secp256k1.json`

### CODE REVIEW

**Comment by sairanjit** [comment]
There are two formats of verification methods for key-type `k256` from test vectors

- https://github.com/w3c-ccg/did-method-key/blob/main/test-vectors/secp256k1.json
- https://github.com/transmute-industries/did-key.js/blob/main/packages/did-key-test-vectors/src/secp256k1/did-key-secp256k1-case-0.json

Which verification method should be used ?
```json
{
  "id": "did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme#zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme",
  "type": "EcdsaSecp256k1VerificationKey2019",
  "controller": "did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme",
  "publicKeyBase58": "23o6Sau8NxxzXcgSc3PLcNxrzrZpbLeBn1izfv3jbKhuv"
}
```
or
```json
{
  "id": "did:key:zQ3shjRPgHQQbTtXyofk1ygghRJ75RZpXmWBMY1BKnhyz7zKp#zQ3shjRPgHQQbTtXyofk1ygghRJ75RZpXmWBMY1BKnhyz7zKp",
  "type": "JsonWebKey2020",
  "controller": "did:key:zQ3shjRPgHQQbTtXyofk1ygghRJ75RZpXmWBMY1BKnhyz7zKp",
  "publicKeyJwk": {
    "kty": "EC",
    "crv": "secp256k1",
    "x": "RwiZITTa2Dcmq-V1j-5tgPUshOLO31FbsnhVS-7lskc",
    "y": "3o1-UCc3ABh757P58gDISSc4hOj9qyfSGl3SGGA7xdc"
  }
}
```

File: packages/core/src/modules/dids/domain/keyDidDocument.ts [hunk 1]
```diff
@@ -27,7 +33,7 @@ const didDocumentKeyTypeMapping: Record<KeyType, (did: string, key: Key) => DidD
   [KeyType.P256]: getJsonWebKey2020DidDocument,
   [KeyType.P384]: getJsonWebKey2020DidDocument,
   [KeyType.P521]: getJsonWebKey2020DidDocument,
-  [KeyType.K256]: getJsonWebKey2020DidDocument,
+  [KeyType.K256]: getSecp256k1DidDoc,
```
Comment (TimoGlastra) [inline packages/core/src/modules/dids/domain/keyDidDocument.ts:36]: Let's use JsonWebKey by default. This seems to be one of the two defaults in DataIntegrity. I think they're moving away from key specific types

**Comment by sairanjit** [comment]
> @sairanjit would be great to get this in for 0.5.0, so if you can update it somewhere today to use `JsonWebKey` that would be great, thanks 🙏

Sure will do it today itself

File: packages/core/src/modules/dids/domain/key-type/secp256k1.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import type { KeyDidMapping } from './keyDidMapping'
+import type { VerificationMethod } from '../verificationMethod'
+
+import { KeyType } from '../../../../crypto/KeyType'
+import { CredoError } from '../../../../error'
+import {
+  VERIFICATION_METHOD_TYPE_ECDSA_SECP256K1_VERIFICATION_KEY_2019,
```
Comment (TimoGlastra) [inline packages/core/src/modules/dids/domain/key-type/secp256k1.ts:7]:
```suggestion
  VERIFICATION_METHOD_TYPE_ECDSA_SECP256K1_VERIFICATION_KEY_2019,
  VERIFICATION_METHOD_TYPE_JSON_WEB_KEY_2020,
```

File: packages/core/src/modules/dids/domain/keyDidDocument.ts (hunk 1 above)
Comment (sairanjit) [reply packages/core/src/modules/dids/domain/keyDidDocument.ts]: - Updated to `JsonWebKey` as default
- Should we later add a way to use different didDoc If someone specifically wants to use them ?

File: packages/core/src/modules/dids/domain/key-type/secp256k1.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  getJsonWebKey2020,
+  getKeyFromEcdsaSecp256k1VerificationKey2019,
+  getKeyFromJsonWebKey2020,
+  isEcdsaSecp256k1VerificationKey2019,
+  isJsonWebKey2020,
+} from '../verificationMethod'
+
+export const keyDidSecp256k1: KeyDidMapping = {
+  supportedVerificationMethodTypes: [VERIFICATION_METHOD_TYPE_ECDSA_SECP256K1_VERIFICATION_KEY_2019],
```
Comment (TimoGlastra) [inline packages/core/src/modules/dids/domain/key-type/secp256k1.ts:16]:
```suggestion
  supportedVerificationMethodTypes: [
    VERIFICATION_METHOD_TYPE_ECDSA_SECP256K1_VERIFICATION_KEY_2019,
    VERIFICATION_METHOD_TYPE_JSON_WEB_KEY_2020,
  ],
```

**Review by TimoGlastra — APPROVED** [review]

File: packages/core/src/modules/dids/domain/keyDidDocument.ts (hunk 1 above)
Comment (TimoGlastra) [reply packages/core/src/modules/dids/domain/keyDidDocument.ts]: Yeah that might be useful -- currently they can call the `getSecpVerificationMethod` and construct a did doc on their own.

**Pull request merged**